charts/datahub/templates/datahub-upgrade/_upgrade.tpl

```diff
@@ -39,15 +39,6 @@ Return the env variables for upgrade jobs
   value: "{{ .Values.global.sql.datasource.url }}"
 - name: EBEAN_DATASOURCE_DRIVER
   value: "{{ .Values.global.sql.datasource.driver }}"
-{{- if .Values.global.datahub.metadata_service_authentication.enabled }}
-- name: DATAHUB_SYSTEM_CLIENT_ID
```

Comment:
I am a bit confused here too why this is causing the problem. 
We have 4 jobs https://github.com/search?q=repo%3Aacryldata%2Fdatahub-helm%20%20include%20%22datahub.upgrade.env%22&type=code which include this `datahub.upgrade.env` and 3 of them are getting this new config. Which means something with `[datahub-system-update-job.yml](https://github.com/acryldata/datahub-helm/blob/30eee991bbad5b398051758c1a97a5d028430427/charts/datahub/templates/datahub-upgrade/datahub-system-update-job.yml#L66)` is causing the problem. 

Now would system update job be able to work without these variables? I am not sure about that.

Reply:
The system update job runs early, before secrets are created. Second, the system update doesn't access any GMS services since it is not running at that early (or running a read-only version of the previous release). The non-existent secrets cannot be mounted in the job. Maybe there is another way to fix it, but this is the easiest, I think.